### PR TITLE
Testing setup

### PR DESCRIPTION
**Pull request by LULIU-9**
Add jest and RTL dependencies into package.json
jest.setup.js and jest.config.js to set up and config jest.
__mock__ file for Stylesheets and images.
__tests__ for test suites.

### CODE REVIEW

File: __mock__/fileMock.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
+    src: '/img.jpg',
```
Comment (tatoMa) [inline __mock__/fileMock.js:2]: So where is this img.jpg file located?

File: __mock__/styleMock.js [hunk 2]
```diff
@@ -0,0 +1 @@
+module.exports = {}
```
Comment (tatoMa) [inline __mock__/styleMock.js:1]: unnecessary file?

File: __tests__/snapshot.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { render } from '@testing-library/react'
+import Home from '../pages/index'
+
+it('renders homepage unchanged', () => {
+  const { container } = render(<Home />)
+  expect(container).toMatchSnapshot()
+})
```
Comment (tatoMa) [inline __tests__/snapshot.js:7]: Nice to have a snapshot test as an example but we don't need it as our regular testing. Unit test is the only one for now.

You can keep this file BTW.

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,21 @@
+const nextJest = require('next/jest')
+
+const createJestConfig = nextJest({
+  // Provide the path to your Next.js app to load next.config.js and .env files in your test environment
+  dir: './',
+})
+
+// Add any custom config to be passed to Jest
+const customJestConfig = {
+  setupFilesAfterEnv: ['<rootDir>/jest.setup.js'],
```
Comment (tatoMa) [inline jest.config.js:10]: Please also add coverageThreshold 90% as our minimum coverage rate.

File: package.json [hunk 5]
```diff
@@ -6,6 +6,7 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
+    "test": "jest --watch",
```
Comment (tatoMa) [inline package.json:9]: This one is for the watch mode of the test. We also need a normal mode test.

Try to create a new script with the command "jest --colors --config jest.config.js"

**Review by tatoMa — APPROVED** [review]
LGTM.

**Comment by tatoMa** [comment]
image

**Review by Cameron933 — APPROVED** [review]
LGTM

**Review by xzci — APPROVED** [review]

**Pull request merged**